packages/calcite-components/src/components/list/list.tsx

```diff
@@ -742,6 +748,51 @@ export class List implements InteractiveComponent, LoadableComponent, SortableCo
     }
   };
 
+  private findAncestorOfFirstFilteredItem = (filteredItems: HTMLCalciteListItemElement[]): void => {
+    if (this.topLevelAncestorsMap.size > 0) {
+      this.topLevelAncestorsMap.forEach(
+        (value: HTMLCalciteListItemElement, key: HTMLCalciteListItemElement) => {
+          value.removeAttribute("data-filter");
+          key.removeAttribute("data-filter");
+        }
+      );
+      this.topLevelAncestorsMap.clear();
+    }
+
+    filteredItems.forEach((item) => {
+      if (item.hasAttribute("data-filter")) {
+        item.removeAttribute("data-filter");
+      }
+    });
+
+    const firstFilteredItem = filteredItems[0];
+    const topLevelAnchor = this.getTopLevelAncestorItemElement(firstFilteredItem);
+    if (topLevelAnchor) {
+      this.topLevelAncestorsMap.set(firstFilteredItem, topLevelAnchor);
+    }
+
+    filteredItems[0].setAttribute("data-filter", "0");
+    this.topLevelAncestorsMap.get(filteredItems[0])?.setAttribute("data-filter", "0");
+  };
+
+  private getTopLevelAncestorItemElement = (
```

Comment:
I think you could use the [`walkUpAncestry`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/utils/dom.ts#L197C23-L197C31) util for this.

Reply:
Per our convo, we can look at updating some of our `dom` utils to help cover this use case later (i.e., recursive `closest` without crossing shadow DOM).